Snowflake.Data/Client/SnowflakeDbCommand.cs

```diff
@@ -11,20 +11,27 @@
 using System.Threading.Tasks;
 using Newtonsoft.Json;
 using Snowflake.Data.Log;
+using System.Text.RegularExpressions;
 
 namespace Snowflake.Data.Client
 {
     [System.ComponentModel.DesignerCategory("Code")]
     public class SnowflakeDbCommand : DbCommand
     {
-        private DbConnection connection;
+        private SnowflakeDbConnection connection;
 
         private SFStatement sfStatement;
 
         private SnowflakeDbParameterCollection parameterCollection;
 
         private SFLogger logger = SFLoggerFactory.GetLogger<SnowflakeDbCommand>();
 
+        // Async max retry and retry pattern
+        private const int AsyncNoDataMaxRetry = 24;
+        private readonly int[] _asyncRetryPattern = { 1, 1, 2, 3, 4, 8, 10 };
+
+        private static readonly Regex UuidRegex = new Regex("^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$");

```

Comment:
Based on [uuid docs](https://www.uuidtools.com/what-is-uuid#:~:text=Are%20UUIDs%20Case%2Dsensitive%3F,UUIDs%20in%20upper%20and%20lowercase.), it's not case sensitive so it can contain capital letters. But at the same time, it says UUID generators should only output in lowercase letters:
```No, UUIDs are written in base 16 which uses numbers 0-9 and characters a-f. There is no distinction between upper and lowercase letters. However, [RCF 4122 section 3](https://tools.ietf.org/html/rfc4122#section-3) requires that UUID generators output in lowercase and systems accept UUIDs in upper and lowercase.```

I'll modify the regex to include uppercase letters